crates/status/src/impl/bitstring_status_list/mod.rs

```diff
@@ -510,45 +672,55 @@ impl StatusList {
         &self,
         id: Option<UriBuf>,
         status_purpose: StatusPurpose,
-        status_message: Vec<StatusMessage>,
     ) -> BitstringStatusList {
-        BitstringStatusList::new(
-            id,
-            status_purpose,
-            self.bit_string.status_size,
-            self.bit_string.encode(),
-            self.ttl,
-            status_message,
-        )
+        BitstringStatusList::new(id, status_purpose, self.bit_string.encode(), self.ttl)
+    }
+
+    pub fn into_unsized(self) -> StatusList {
+        StatusList {
+            bit_string: self.bit_string.into_unsized(),
+            ttl: self.ttl,
+        }
     }
 }
 
 pub struct BitStringIter<'a> {
     bit_string: &'a BitString,
+    status_size: StatusSize,
     index: usize,
 }
 
 impl<'a> Iterator for BitStringIter<'a> {
     type Item = u8;
 
     fn next(&mut self) -> Option<Self::Item> {
-        self.bit_string.get(self.index).map(|status| {
-            self.index += 1;
-            status
-        })
+        self.bit_string
+            .get(self.status_size, self.index)
+            .map(|status| {
+                self.index += 1;
+                status
+            })
     }
 }
 
 impl StatusMap for StatusList {
```

Comment:
Is `StatusMap` mainly a convenience interface for interacting with both types of status lists (i.e. does not need to strictly conform to the specs)? If so, wondering if it makes sense to implement the common for the `SizedStatusList` instead of the `BitstringStatusList`, since it would be safer to interact with & clean up the interface?

Reply:
But that's the issue, we wouldn't be able to build a `SizedStatusList` from a status list credential, because the `statusSize` property isn't there anymore :sweat:

Reply:
Sorry I don't think I follow 😅 isn't it just about implementing this interface for `StatusList` (or `SizedStatusList`)? Why not remove the `impl StatusMap for StatusList` entirely? Maybe I just don't understand where the `StatusMap` is used normally or what you mean

Reply:
discussed -- useful for testing, and maybe not worth modifying it now while the spec is still in flux